borg/hashindex.pyx

```diff
@@ -158,10 +159,15 @@ cdef class NSKeyIterator:
         self.key = hashindex_next_key(self.index, <char *>self.key)
         if not self.key:
             raise StopIteration
-        cdef int32_t *value = <int32_t *>(self.key + self.key_size)
+        cdef uint32_t *value = <uint32_t *>(self.key + self.key_size)
         return (<char *>self.key)[:self.key_size], (_le32toh(value[0]), _le32toh(value[1]))
 
-assert INT32_MAX == 2**31-1
+
+assert UINT32_MAX == 2**32-1
+
+# module-level constant because cdef's in classes can't have default values
+cdef uint32_t _MAX_REF = 2**32-3
+MAX_REF = _MAX_REF
```

Comment:
maybe we could reserve a few more "special" values?
you never know when you need them...